CSharpSourceCode/AttributeDataSystem/PartyAttributeManager.cs

```diff
@@ -0,0 +1,439 @@
+            
+                text += " are supporting the defenders(" + _currentPlayerEvent.DefenderSide.Parties.Count+")";
+                
+                _activePartyAttributes = ActivePartyAttributes;
+                BattleAttributesArgs e = new BattleAttributesArgs()
+                {
+                    activeParties = ActivePartyAttributes
+                };
+                
+                NotifyBattlePartyObservers?.Invoke(this,e);
+            }
+        }
+        
+        private void OnBattleStarted(
+            PartyBase attackerParty,
+            PartyBase defenderParty,
+            object subject,
+            bool showNotification)
+        {
+            PartyAttribute attackPartyPartyAttribute= GetAttribute(attackerParty.Id.ToString());
+            PartyAttribute defenderPartyPartyAttribute= GetAttribute(defenderParty.Id.ToString());
+            if (attackPartyPartyAttribute != null || defenderPartyPartyAttribute != null)
+            {
+             //   TOWCommon.Say(attackPartyWorldMapAttribute.id+ "("+attackPartyWorldMapAttribute.culture+")"+ " is fighting now " + defenderPartyWorldMapAttribute.id+ "("+defenderPartyWorldMapAttribute.culture+")");
+            }
+                
+        }
+
+        public  PartyAttribute GetAttribute(string id)
+        {
+            if (_partyAttributes.ContainsKey(id))
+            {
+                return _partyAttributes[id];
+            }
+            else
+            {
+                TOWCommon.Say("couldnt find party");
+                return null;
+            }
+            
+        }
+        
+        private void EnterPartyIntoDictionary(MobileParty party)
+        {
+            //filling fresh created parties with data
+            
+            if (_partyAttributes.ContainsKey(party.Party.Id))
+            {
+                TOWCommon.Say(party.Id+  " was already there");
+                return;
+            }
+            
+            PartyAttribute partyAttribute = new PartyAttribute();
+            partyAttribute.id = party.Party.Id;
+            partyAttribute.PartyBase = party.Party;
+            
+            foreach (var troop in party.Party.MemberRoster.GetTroopRoster())
+            {
+                if(troop.Character.IsHero)
+                    continue;
+                
+                StaticAttribute staticAttribute = new StaticAttribute();
+                staticAttribute.race = troop.Character.Culture.StringId;
+                staticAttribute.status = "battle ready";
+                staticAttribute.MagicEffects = new List<string>();
+                staticAttribute.id = troop.ToString();
+                staticAttribute.number = party.Party.MemberRoster.Count;
+                partyAttribute.RegularTroopAttributes.Add(staticAttribute);
+            }
+
+            if (party.IsBandit)
+            {
+                StaticAttribute staticAttribute = new StaticAttribute();
+                staticAttribute.race = "bandit";
```

Comment:
the static attribute to each unit which origins from a troop tree, or a hero or companion. in case of bandits currently, they only have a single static attribute, since I couldn't access its troop tree.